subsys/net/lib/nrf_cloud/Kconfig

```diff
@@ -23,6 +23,8 @@ rsource "Kconfig.nrf_cloud_alert"
 
 rsource "Kconfig.nrf_cloud_log"
 
+rsource "Kconfig.nrf_cloud_coap"
+
```

Comment:
Adding a comment here because I can't add it to the correct line, but if `NRF_CLOUD_REST` does not need to be enabled to use CoAP, `NRF_CLOUD_COAP` should be added to line 34.

Reply:
Whilst on the topic. When enabling NRF_CLOUD_COAP without NRF_CLOUD_REST I get some strange error log output:

```
zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj -MF zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj -c /home/sir1/dev/ncs/nrf/samples/net/mqtt/build/zephyr/misc/generated/configs.c
/tmp/ccdh6TUN.s: Assembler messages:
/tmp/ccdh6TUN.s:903: Error: missing expression
[1053/1268] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/net/lib/coap/coap.c.obj
ninja: build stopped: subcommand failed.
```

Reply:
The current commits (which include some of the changes you all have suggested) build fine without `NRF_CLOUD_REST`, and with `NRF_CLOUD_APGS=n`, and `NRF_CLOUD_PGPS=n`.

Reply:
I think `NRF_CLOUD_COAP` should still be added to the condition on Kconfig line 34. Otherwise compilation fails unless `MODEM_JWT` or `NRF_CLOUD_REST` is also enabled.